src/mempool/mempool.rs

```diff
@@ -25,40 +27,44 @@ impl Default for Config {
     }
 }
 
-#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
-pub struct MempoolKey {
-    inserted_at: Instant,
-}
-
 pub struct MempoolMessagesRequest {
     pub shard_id: u32,
     pub message_tx: oneshot::Sender<Vec<MempoolMessage>>,
     pub max_messages_per_block: u32,
 }
 
 pub struct Mempool {
+    capacity_per_shard: usize,
     shard_stores: HashMap<u32, Stores>,
     message_router: Box<dyn MessageRouter>,
     num_shards: u32,
     mempool_rx: mpsc::Receiver<MempoolMessage>,
     messages_request_rx: mpsc::Receiver<MempoolMessagesRequest>,
-    messages: HashMap<u32, BTreeMap<MempoolKey, MempoolMessage>>,
+    messages: HashMap<u32, BTreeMap<String, MempoolMessage>>,
```

Comment:
This changes the ordering in which messages are pulled from the mempool, if the ordering is important (e.g. if we want FIFO) we may need to find another way to choose messages from the pool